server/bin/pbench-server-prep-shim-002.py

```diff
@@ -0,0 +1,225 @@
+                "os.mkdir: Unable to create errors destination directory: {}".format(errors))
+            return 1
+
+    list_check = list()
+    receive_dir_check = glob.iglob(os.path.join(receive_dir,'**'), recursive=True)
+    for tar in receive_dir_check:
+        if tar.endswith('.tar.xz.md5'):
+            list_check.append(tar)
+
+    list_check.sort()
+    logger.info(config.TS)
+
+    ntotal = ntbs = nerrs = nquarantined = ndups = 0
+
+    for tbmd5 in list_check:
+        ntotal = ntotal+1
+        tb = tbmd5[0:-4]
+        tbdir = os.path.dirname(tb)
+        result_name = os.path.basename(tb)
+        resultname = result_name[0:-7]
+
+        controller = os.path.basename(tbdir)
+        dest = os.path.join(config.ARCHIVE, controller)
+
+        if os.path.isfile(os.path.join(dest, result_name)) and os.path.isfile(os.path.join(dest, os.path.basename(tbmd5))) and os.path.exists(os.path.join(dest, result_name)) and os.path.exists(os.path.join(dest, os.path.basename(tbmd5))):
+            logger.error("{}: Duplicate: {} duplicate name".format(config.TS, tb))
+            quarantine(os.path.join(duplicates,controller), tb, tbmd5)
+            ndups = ndups+1
+            continue
+
+        # read the md5sum from md5 file
+        try:
+            with open(tbmd5) as f:
+                archive_md5_hex_value = f.readline().split(" ")[0]
+        except Exception:
+            logger.error("Quarantine: Could not read {}".format(tb))
+            quarantine(os.path.join(qdir_md5,controller), tb, tbmd5)
+            nquarantined = nquarantined+1
+            continue
+
+        # match md5sum of the tarball to its md5 file
+        try:
+            archive_tar_hex_value = md5sum(tb)
+        except Exception:
+            logger.error("Quarantine: Could not read {}".format(tb))
+            quarantine(os.path.join(qdir_md5,controller), tb, tbmd5)
+            nquarantined = nquarantined+1
+            continue
+
+        if archive_tar_hex_value != archive_md5_hex_value:
+            logger.error("{}: Quarantined: {} failed MD5 check".format(config.TS, tb))
+            quarantine(os.path.join(qdir_md5,controller), tb, tbmd5)
+            nquarantined = nquarantined+1
+            continue
+       
+        if not os.path.isdir(os.path.join(dest,'TODO')):
+            try:
+                os.makedirs(os.path.join(dest,'TODO'))
+            except FileExistsError:
+                # directory already exists, ignore
+                pass
+            except Exception:
+                logger.error("{}: Error in creating TODO directory. {}".format(config.TS, e))
+                quarantine(os.path.join(errors,controller), tb, tbmd5)
+                nerrs = nerrs+1
+                continue
+
+        try:
+            shutil.copy(tb, dest)
+            shutil.copy(tbmd5, dest)
+        except Exception:
+            logger.error("{}: Error in copying tarball files to Destination path. {}".format(config.TS, e))
```

Comment:
you can use logger.exception which will print out the `e` variable without you needing to reference it.

Reply:
ok